### PR TITLE
[TensorV2] Add Transpose Support for Tensor

### PR DESCRIPTION
**Pull request by djeong20**
This pull request adds the ability to transpose tensors, which allows users to flip or rotate their data along its axes.

**Changes proposed in this PR:**
- Added a new method called transpose() to the TensorV2 class, which takes direction argument: the axes to be swapped.

**Self-evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test:   [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/tensor/half_tensor.cpp [hunk 1]
```diff
@@ -565,6 +539,98 @@ TensorV2 &HalfTensor::divide(TensorV2 const &m, TensorV2 &output) const {
...
+  if (!contiguous) {
+    throw std::runtime_error("Cannot copy non-contiguous tensor");
+  }
+
+  if (size() != from.size())
+    throw std::invalid_argument("Size of tensor to copy must match");
+
+  switch (from.getDataType()) {
+  case ml::train::TensorDim::DataType::FP32:
+    scopy(size(), from.getData<float>(), 1, (_FP16 *)getData(), 1);
```
Comment (skykongkong8) [inline nntrainer/tensor/half_tensor.cpp:557]: Using swich-case instead of previous if-else seems much better!

**Review by skykongkong8 — APPROVED** [review]

**Review by DonghakPark — APPROVED** [review]
LGTM!

**Review by jihochu — APPROVED** [review]
LGTM

File: nntrainer/tensor/tensor_base.h [hunk 2]
```diff
@@ -21,6 +21,34 @@
 #include <tensor_dim.h>
 #include <util_func.h>
 
+#define transposeloop(cl, ci, cj, ck, sl, si, sj, sk)                 \
```
Comment (jijoongmoon) [inline nntrainer/tensor/tensor_base.h:24]: I'm not sure if it is working with omp directives in define. If it can, then we could try to apply it.

**Review by jijoongmoon — APPROVED** [review]
LGTM except minor comment.

**Comment by djeong20** [comment]
Currently, the TAOS/CI clang-format version differs from Git action static checks. there's a mismatch that causes clang-format style error (the behavior of the clang-format varies for different versions). It is ok to merge.

File: nntrainer/tensor/tensor_base.h (hunk 2 above)
Comment (djeong20) [reply nntrainer/tensor/tensor_base.h:24]: agree. I think we should test it after when openMP is enabled. I'll keep it in mind and come back to this later.

**Pull request merged**